test/function_wrappers.ts

```diff
@@ -77,20 +77,23 @@ describe('wrapUserFunction', () => {
   it('correctly wraps an async CloudEvent function', done => {
     const request = createRequest(CLOUD_EVENT);
     const response = createResponse();
-    const func = wrapUserFunction(async (cloudEvent: CloudEvent) => {
-      assert.deepStrictEqual(cloudEvent, CLOUD_EVENT);
-      // await to make sure wrapper handles async code
-      await new Promise(resolve => setTimeout(resolve, 20));
-      done();
-    }, 'cloudevent');
+    const func = wrapUserFunction(
+      async (cloudEvent: CloudEvent<Record<string, string>>) => {
```

Comment:
hm, well I guess I didn't add a default type for the `CloudEvent` interface so if you used it you need to provide a template type. We could provide one, but that would be effectively creating our own type that extends the cesdk object with a default type.

Reply:
a few thoughts here:

1.  I think PR #391 was actually a breaking change because it renamed our exported type from `CloudEventContext` => `CloudEvent`. I will mark this PR as breaking so release please does the right thing before we publish a new version.
2.  I am not too worried about the changes in this file because `wrapUserFunction` isn't part our public API so the changes to it aren't really breaking.
3. We can definitely modify the `CloudEvent` interface exported by `functions.ts` to make the template type optional:

```typescript
import {CloudEventV1} from 'cloudevents';

/**
 * @public
 */
export interface CloudEvent<T=unknown> extends CloudEventV1<T> {};
```

The only down side I can think of is that we loose the top level JSDOC comment. I am not sure if this is actually a better user experience than requiring TS developers to provide the template type...

(I actually think 3 might be something we should plumb back to the CESDK if we think it is the right thing to do).

Reply:
Another idea if we want to fully eliminate all breaking changes from this PR and #391 is to keep around the old export name:

```typescript
import {CloudEventV1 as CloudEvent} from 'cloudevents';

/**
 * @public
 */
export {CloudEvent};

/**
 * Add this type back to eliminate breaking changes.
 * @public
 */
export interface CloudEventContext extends CloudEventV1<
  // what ever we had originally typed data as...
  string|number|unknown
> {};
```

Reply:
Yes, I hear you. Can we have a separate test case for correctly wrapping a typed Cloud Function?

- `'correctly wraps an async CloudEvent function`
- `'correctly wraps an async typed CloudEvent function` (suggestion)

This means don't modify this test case. Copy can create a new test case (suggestion).

Reply:
Currently `wrapUserFunction` requires a type annotation there or there is a "has implicit type any' error

Reply:
Discussed offline. We will keep the template type as required for now. We can make it optional later in a non-breaking change if we decide that is better.